openid-4-verifiable-presentations-1_0.md

```diff
@@ -432,54 +427,43 @@ Note: "https://self-issued.me/v2" is a symbolic string and can be used as an `au
 
-* `x509_san_uri`: When the Client Identifier Scheme is `x509_san_uri`, the Client Identifier MUST be a URI and match a `uniformResourceIdentifier` Subject Alternative Name (SAN) [@!RFC5280] entry in the leaf certificate passed with the request. The request MUST be signed with the private key corresponding to the public key in the leaf X.509 certificate of the certificate chain added to the request in the `x5c` JOSE header [@!RFC7515] of the signed request object. The Wallet MUST validate the signature and the trust chain of the X.509 certificate. All Verifier metadata other than the public key MUST be obtained from the `client_metadata` parameter. If the Wallet can establish trust in the Client Identifier authenticated through the certificate, e.g. because the Client Identifier is contained in a list of trusted Client Identifiers, it may allow the client to freely choose the `redirect_uri` value. If not, the `redirect_uri` value MUST match the Client Identifier.
+If multiple of the above mechanisms are supported and apply, the Wallet SHOULD use the following order:
+- If the request is signed, 
+  - Pre-registered Clients
+  - X.509 Certificates
+  - Verifier Attestations
+  - DID
+  - OpenID Federation
+  - Other specifications or ecosystem-specific support
+- If the request is unsigned,
+  - Pre-registered Clients: In addition to the rules above for pre-registered Clients, the Wallet MUST check that `require_request_object_signing` is set to `false` or omitted from the pre-registered Client metadata. The request MUST be rejected if `require_request_object_signing` is `true`, otherwise processing continues.
+  - Client ID matches the Redirect and/or Response URI
 
-To use `client_id_scheme` values `entity_id`, `did`, `verifier_attestation`, `x509_san_dns`, and `x509_san_uri`, Verifiers MUST be confidential clients. This might require changes to the technical design of native apps as such apps are typically public clients.
+To prevent downgrade attacks, the Wallet MUST cyrptographically bind the JWK Thumbprint of the public key that was used to verify the signature of the signed request, to the Verifiable Presentation(s) in the response. The exact mechanism of how the JWK Thumbprint is bound to the Verifiable Presentation(s) is defined per Credential Format. For example, for SD-JWT VC, the JWK Thumbprint is prepended to the `client_id` value and included in the `aud` claim of the the Key Binding JWT, e.g., `aud=<JWK-Thumbprint>:<client_id>`.
 
-Other specifications can define further values for the `client_id_scheme` parameter. It is RECOMMENDED to use collision-resistant names for such values.
+If the Verifier provided a signed request to the Wallet, the Verifier MUST verify the JWK Thumbprint included in the Verifiable Presentation(s) of the response corresponds to the key that signed the original request. The response MUST be rejected if verification fails, otherwise processing continues.
```

Comment:
This section confuses me a bit, but I am easily confused. Seems a bit like "can you repeat to me what I said so I am sure you understood it".

Am a getting it right? I might be missing a smallish example of possible vectors and surfaces that we are trying to mitigate i.e. the reasoning behind this "proof of computation".